File: backend/app/schemas/onboarding.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class OnboardingCreate(BaseModel):
    cleanliness: str
    dishes: str
    common_objects: str
    noise: str
    visits: str
    sleepovers: str
    wake_up: str
    night_noise: str
    smoking: str
    alcohol: str
    pets: str
    bills: str
    food: str
    communication: str
    conflicts: str
    rules: str
    culture: str
    space: str
    lifestyle: str


class OnboardingResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID

    cleanliness: str
    dishes: str
    common_objects: str
    noise: str
    visits: str
    sleepovers: str
    wake_up: str
    night_noise: str
    smoking: str
    alcohol: str
    pets: str
    bills: str
    food: str
    communication: str
    conflicts: str
    rules: str
    culture: str
    space: str
    lifestyle: str

    created_at: datetime
    updated_at: datetime

File: frontend/src/types/onboarding.ts
export interface OnboardingAnswers {
  cleanliness: string;
  dishes: string;
  common_objects: string;
  noise: string;
  visits: string;
  sleepovers: string;
  wake_up: string;
  night_noise: string;
  smoking: string;
  alcohol: string;
  pets: string;
  bills: string;
  food: string;
  communication: string;
  conflicts: string;
  rules: string;
  culture: string;
  space: string;
  lifestyle: string;
}

export interface OnboardingResponse extends OnboardingAnswers {
  id: string;
  user_id: string;
  created_at: string;
  updated_at: string;
}
